python/ray/tests/test_runtime_env_packaging.py

```diff
@@ -153,6 +162,53 @@ def test_create_upload_once(self, tmp_path, random_dir, ray_start_regular):
         assert uploaded
 
 
+class TestStorePackageInGcs:
+    class DisconnectedClient(GcsClient):
+        """Mock GcsClient that fails cannot put in the GCS."""
+
+        def __init__(self, *args, **kwargs):
+            pass
+
+        def internal_kv_put(self, *args, **kwargs):
+            raise RuntimeError("Cannot reach GCS!")
+
+    def raise_runtime_error(self, *args, **kwargs):
+        raise RuntimeError("Raised a runtime error!")
+
+    def test_upload_succeeds(self, ray_start_regular):
+        """Check function behavior when upload succeeds."""
+
+        uri = "gcs://test.zip"
+        bytes = b"test"
+
+        assert len(bytes) < GCS_STORAGE_MAX_SIZE
+        assert not _internal_kv_exists(uri, namespace=KV_NAMESPACE_PACKAGE)
+        assert _store_package_in_gcs(uri, bytes) == len(bytes)
+        assert bytes == _internal_kv_get(uri, namespace=KV_NAMESPACE_PACKAGE)
+
+    def test_upload_fails(self):
+        """Check that function throws useful error when upload fails."""
+
+        uri = "gcs://test.zip"
+        bytes = b"test"
+
+        assert len(bytes) < GCS_STORAGE_MAX_SIZE
+
+        _internal_kv_reset()
+        _initialize_internal_kv(self.DisconnectedClient())
+        with pytest.raises(RuntimeError, match="Failed to store package in the GCS"):
+            _store_package_in_gcs(uri, bytes)
+
+    def test_package_size_too_large(self):
+        """Check that function throws useful error when package is too large."""
+
+        uri = "gcs:/test.zip"
```

Comment:
```suggestion
        uri = "gcs://test.zip"
```
Just in case it fails some URI parsing now or in the future